src/GiraffeGenerator.Core/CodeGen.fs

```diff
@@ -166,12 +66,149 @@ let giraffeAst (api: Api) =
+                                     )
+                                 |> Seq.groupBy fst
+                                 |> Seq.map (fun (l, v) -> l, v |> Seq.map snd |> Seq.exactlyOne)
+                                 |> Map
+                              (method.Method, method.Name), locationNameMapping
+                          
+              } |> Map
+          
+          let combinedInputTypeNames =
+              api.Paths
+              |> Seq.collect (fun x -> x.Methods)
+              |> Seq.choose
+                  (
+                      fun m ->
+                          m.Parameters
+                          |> Option.filter
+                              (
+                                  fun p ->
+                                      p
+                                      |> Map.toSeq
+                                      |> Seq.map fst
+                                      |> Seq.filter isNotBody
+                                      |> Seq.length > 1
+                              )
+                          |> Option.map (fun _ -> (m.Method, m.Name), requestCommonInputTypeName m)
+                  )
+              |> Map
+
+          let tmpBindingSchemas =
+              [ for path in api.Paths do
+                  for method in path.Methods do
+                      if method.Parameters.IsSome then
+                        for KeyValue(source, schema) in method.Parameters.Value do
+                            if source = Query || not (isNotBody source) then // non-query and non-body bindings don't support default values
+                                let generatedTypes = generateOptionalType schema.Kind schema.DefaultValue (Some schema.Name)
+                                yield! generatedTypes ]
+          let tmpBindingSchemasMap =
+              tmpBindingSchemas
+              |> Seq.collect (fun v -> [ SourceType v.OriginalName, v; GeneratedType v.GeneratedName, v ])
+              |> Map
 
           let allSchemas =
-              [ for path in api.Paths do
+              [ 
+                yield! api.Schemas
+                yield! tmpBindingSchemas |> Seq.map ^ fun v -> { Kind = v.Generated; Name = v.GeneratedName; Docs = None; DefaultValue = None }
+                for path in api.Paths do
                   for method in path.Methods do
-                      if method.Parameters.IsSome then method.Parameters.Value
-                yield! api.Schemas ]
+                      if method.Parameters.IsSome then
+                        for KeyValue(_, schema) in method.Parameters.Value do
+                            schema
+                        if method.Parameters.Value |> Map.toSeq |> Seq.map fst |> Seq.filter isNotBody |> Seq.length > 1 then
+                            let mapping =
+                                parametersLocationNameMapping
+                                |> Map.find (method.Method, method.Name)
```

Comment:
Yes. It would result in empty sub-map if `method.Parameters = Some Map.empty`. It could fail if `method.Parameters = None`, but this is checked before call to `Map.find` (line 177)